controllers/admin/AdminCartRulesController.php

```diff
@@ -762,4 +768,60 @@ public function displayAjaxSearchCartRuleVouchers()
         }
         echo json_encode(['found' => $found, 'vouchers' => $vouchers]);
     }
+
+    /**
+     * For the listing, Override the method displayDeleteLink for the HelperList
+     * That allows to have links with all characters (like < & >)
+     *
+     * @param string $token
+     * @param string $id
+     * @param string|null $name
+     *
+     * @return string
+     */
+    public function displayDeleteLink(string $token, string $id, ?string $name = null): string
+    {
+        if (!$this->access('delete')) {
+            return '';
+        }
+
+        $tpl = $this->createTemplate('helpers/list/list_action_delete.tpl');
+
+        if (!array_key_exists('Delete', self::$cache_lang)) {
+            self::$cache_lang['Delete'] = $this->trans('Delete', [], 'Admin.Actions');
+        }
+        if (!array_key_exists('DeleteItem', self::$cache_lang)) {
+            self::$cache_lang['DeleteItem'] = $this->trans('Delete selected item?', [], 'Admin.Notifications.Info');
+        }
+        if (!array_key_exists('Name', self::$cache_lang)) {
+            self::$cache_lang['Name'] = $this->trans('Name:', [], 'Admin.Shipping.Feature');
+        }
+
+        if (null !== $name) {
+            $name = '\n\n' . self::$cache_lang['Name'] . ' ' . $name;
+        }
+
+        $data = [
+            $this->identifier => $id,
+            'href' => $this->context->link->getAdminLink(
+                'AdminCartRules',
+                true, [],
+                [
+                    'id_cart_rule' => (int) $id,
+                    'deletecart_rule' => 1,
+                ]
+            ),
+            'action' => self::$cache_lang['Delete'],
+        ];
+
+        if ($this->specificConfirmDelete !== false) {
+            $data['confirm'] = null !== $this->specificConfirmDelete
+                ? '\r' . $this->specificConfirmDelete
```

Comment:
What's the aim of using carriage return? (`\r`)

Reply:
It's a `copy/paste` from `classes/helper/HelperList.php`.